### ISSUE TITLE
App crashed when check update

### ISSUE THREAD

**Issue by evgenykushnarenko** [Tobi823/ffupdater#491]
Stacktrace:
```
java.lang.IllegalStateException: Check failed.
	at de.marmaro.krt.ffupdater.network.fdroid.FdroidConsumer.getLatestUpdate(FdroidConsumer.kt:82)
	at de.marmaro.krt.ffupdater.network.fdroid.FdroidConsumer.getLatestUpdate(FdroidConsumer.kt:24)
	at de.marmaro.krt.ffupdater.network.fdroid.FdroidConsumer$getLatestUpdate$1.invokeSuspend(FdroidConsumer.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7918)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@ca52e4d, Dispatchers.Main]

```
Logs:
```
java.lang.IllegalStateException: Check failed.
	at de.marmaro.krt.ffupdater.network.fdroid.FdroidConsumer.getLatestUpdate(FdroidConsumer.kt:82)
	at de.marmaro.krt.ffupdater.network.fdroid.FdroidConsumer.getLatestUpdate(FdroidConsumer.kt:24)
	at de.marmaro.krt.ffupdater.network.fdroid.FdroidConsumer$getLatestUpdate$1.invokeSuspend(FdroidConsumer.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7918)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@ca52e4d, Dispatchers.Main]

```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 78.1.5 (154) release F-Droid/other |
| Device | Pixel 7 Pro (cheetah, cheetah, cheetah) |
| Manufacturer | google (Google) | 
| Supported ABIs | arm64-v8a |
| Android version | 13 (SDK: 33) |
| OS | abfarm-release-2004-0093, android-build, release-keys, 1685043207000 |         

**Describe the bug**

**To Reproduce**

**Expected behavior**

**Error message from FFUpdater**
Please copy and paste the error message with the stacktrace and error log here:

**Comment by aaron1857** [comment]
Similar issue for me. The app crashes when I try to open it. I'm using a Google Pixel 6 with Graphene OS. I'm also using the latest version found in f droid. 

Crash report
`Stacktrace:
```
java.lang.IllegalStateException: Check failed.
	at de.marmaro.krt.ffupdater.network.fdroid.FdroidConsumer.getLatestUpdate(FdroidConsumer.kt:82)
	at de.marmaro.krt.ffupdater.network.fdroid.FdroidConsumer.getLatestUpdate(FdroidConsumer.kt:24)
	at de.marmaro.krt.ffupdater.network.fdroid.FdroidConsumer$getLatestUpdate$1.invokeSuspend(FdroidConsumer.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7940)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ExecInit.main(ExecInit.java:49)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:355)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@77b58c6, Dispatchers.Main]

```
Logs:
```
java.lang.IllegalStateException: Check failed.
	at de.marmaro.krt.ffupdater.network.fdroid.FdroidConsumer.getLatestUpdate(FdroidConsumer.kt:82)
	at de.marmaro.krt.ffupdater.network.fdroid.FdroidConsumer.getLatestUpdate(FdroidConsumer.kt:24)
	at de.marmaro.krt.ffupdater.network.fdroid.FdroidConsumer$getLatestUpdate$1.invokeSuspend(FdroidConsumer.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7940)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ExecInit.main(ExecInit.java:49)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:355)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@77b58c6, Dispatchers.Main]

```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 78.1.5 (154) release F-Droid/other |
| Device | Pixel 6 (oriole, oriole, oriole) |
| Manufacturer | google (Google) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 13 (SDK: 33) |
| OS | grapheneos, grapheneos, release-keys, 1693982059000 |         `

**Comment by yutayu** [comment]
Stacktrace:
```
java.lang.IllegalStateException: Check failed.
	at de.marmaro.krt.ffupdater.network.fdroid.FdroidConsumer.getLatestUpdate(FdroidConsumer.kt:82)
	at de.marmaro.krt.ffupdater.network.fdroid.FdroidConsumer.getLatestUpdate(FdroidConsumer.kt:24)
	at de.marmaro.krt.ffupdater.network.fdroid.FdroidConsumer$getLatestUpdate$1.invokeSuspend(FdroidConsumer.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7406)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:944)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@43d595b, Dispatchers.Main]

```
Logs:
```
java.lang.IllegalStateException: Check failed.
	at de.marmaro.krt.ffupdater.network.fdroid.FdroidConsumer.getLatestUpdate(FdroidConsumer.kt:82)
	at de.marmaro.krt.ffupdater.network.fdroid.FdroidConsumer.getLatestUpdate(FdroidConsumer.kt:24)
	at de.marmaro.krt.ffupdater.network.fdroid.FdroidConsumer$getLatestUpdate$1.invokeSuspend(FdroidConsumer.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7406)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:944)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@43d595b, Dispatchers.Main]

```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 78.1.5 (154) release F-Droid/other |
| Device | LAVIE Tab E 10FHD1 (LAVIETabE10FHD1, LAVIETabE10FHD1, LAVIETabE10FHD1) |
| Manufacturer | NEC (NEC) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 10 (SDK: 29) |
| OS | cm-build-V2, cm, release-keys, 1624263641000 |

**Comment by ayac** [comment]
Stacktrace:
```
java.lang.IllegalStateException: Check failed.
	at de.marmaro.krt.ffupdater.network.fdroid.FdroidConsumer.getLatestUpdate(FdroidConsumer.kt:25)
	at de.marmaro.krt.ffupdater.network.fdroid.FdroidConsumer$getLatestUpdate$1.invokeSuspend(FdroidConsumer.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7050)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@d77cab, Dispatchers.Main]
Timestamp: 2023-09-08T18:48:27.992

```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 77.7.14 (140) release |
| Device | SM-G950F (dreamltexx, dreamlte, universal8895) |
| Manufacturer | samsung (samsung) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 9 (SDK: 28) |
| OS | SWDH4607, dpi, release-keys, 1658738761000 |         

Start crashing today also on other devices i own with  FFUpdater updated to the latest version 78.1.5 .

**Comment by Tobi823** [comment]
Bug is now fixed (in the version 78.1.6)

**Comment by yutayu** [comment]
Thanks :)
